CRM/Contribute/Form/Task/PDF.php

```diff
@@ -187,6 +195,14 @@ public function postProcess() {
       $objects['contribution']->receive_date = CRM_Utils_Date::isoToMysql($objects['contribution']->receive_date);
 
       $values = array();
+      if (isset($params['fromEmailAddress']) && !$elements['createPdf']) {
+        $fromEmail = $params['fromEmailAddress'];
+        $from = CRM_Utils_Array::value($fromEmail, $this->_emails);
+        $fromDetails = explode(' <', $from);
```

Comment:
Also $params['fromEmailAddress'] outputs as the value id for the relevant option value when you select that is from the list of from email addresses